django/db/backends/oracle/operations.py

```diff
@@ -310,17 +312,88 @@ def savepoint_create_sql(self, sid):
     def savepoint_rollback_sql(self, sid):
         return convert_unicode("ROLLBACK TO SAVEPOINT " + self.quote_name(sid))
 
+    @cached_property
```

Comment:
still, a method with a name like this will be used in normal migration operations implementation and then break in surprising ways. At least rename it something like "_foreign_key_constraints_for_test_flush" (or something less ugly but still clear) and add a comment about its unsuitability for use in `SchemaEditor`